src/cpp/KnowledgeItem.h

```diff
@@ -29,6 +30,7 @@ class KnowledgeItem {
 
 	virtual void save_to(BaseStorageSaver& storage);
 
+    virtual double match(SearchCriteria<std::string, std::time_t> *criteria);

```

Comment:
Why are we passing pointer to SearchCriteria here?
Is SearchCriteria expected to be base class for different subclasses in future?